### PR TITLE
Run heap parsing tests on different platforms 

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Running https://github.com/rizinorg/rizin/pull/1109 with Travis and AppVeyor 

**Test plan**

CI is green

cc @PulakIIIT

### CODE REVIEW

**Comment by XVilka** [comment]
@PulakIIIT fails to build on Windows with MSVC: https://ci.appveyor.com/project/rizinorg/rizin/builds/39130526/job/43br45k0ede415jl

```
..\librz\include\rz_heap_jemalloc.h(4): fatal error C1083: Cannot open include file: 'rz_jemalloc/internal/jemalloc_internal.h': No such file or directory
```

**Comment by MalhotraPulak** [comment]
`rz_jemalloc/internal/jemalloc_internal.h` refers to a file located inside the `shlr` directory. This file is being used by the Jemalloc implementation. I am not sure why this error is happening on windows and not other platforms. Maybe it is related to the build process or the compiler. I don't know how to approach fixing this error at the moment.

**Comment by XVilka** [comment]
It might be related to some meson options:
- https://github.com/rizinorg/rizin/blob/dev/meson.build#L304
- https://github.com/rizinorg/rizin/search?q=HAVE_JEMALLOC

**Comment by MalhotraPulak** [comment]
@XVilka I fixed the initial issue but some Jemalloc files are using the `strings.h` header which is not present in Windows. Should I add an implementation of the header to the codebase [like this one](https://github.com/win32ports/strings_h/blob/master/strings.h)?

**Comment by XVilka** [comment]
@PulakIIIT I recommend to just substitute required functions with ones from `rz_util.h` instead. Since we plan to do a Jemalloc code refactoring anyway, I think it's completely fine.

**Comment by MalhotraPulak** [comment]
@XVilka  I implemented the required [`ffs`](https://en.wikipedia.org/wiki/Find_first_set)  from function from `strings.h` as `rz_num_ffs` and used that in the Jemalloc files. This messed up things a lot and now I'm getting a really long error [log](https://ci.appveyor.com/project/rizinorg/rizin/builds/39210968/job/xt7lccm6kbpl7shc).  The code is compiling fine on other platforms still.

**Comment by XVilka** [comment]
This is the root of the error:
```
c:\projects\rizin\shlr\heap\include\rz_jemalloc\internal\jemalloc_internal_decls.h(57): warning C4005: 'restrict': macro redefinition
..\librz\include\rz_types.h(159): note: see previous definition of 'restrict'
c:\projects\rizin\shlr\heap\include\rz_jemalloc\internal\../jemalloc.h(198): error C2375: 'malloc': redefinition; different linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_malloc.h(97): note: see declaration of 'malloc'
c:\projects\rizin\shlr\heap\include\rz_jemalloc\internal\../jemalloc.h(201): error C2375: 'calloc': redefinition; different linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_malloc.h(62): note: see declaration of 'calloc'
c:\projects\rizin\shlr\heap\include\rz_jemalloc\internal\../jemalloc.h(210): error C2375: 'realloc': redefinition; different linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_malloc.h(122): note: see declaration of 'realloc'
c:\projects\rizin\shlr\heap\include\rz_jemalloc\internal\../jemalloc.h(212): error C2375: 'free': redefinition; different linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_malloc.h(85): note: see declaration of 'free'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\TlHelp32.h(153): error C2059: syntax error: '('
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\TlHelp32.h(156): error C2059: syntax error: '}'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\TlHelp32.h(157): error C2143: syntax error: missing '{' before '*'
```

File: librz/util/unum.c [hunk 1]
```diff
@@ -913,3 +913,13 @@ RZ_API double rz_num_cos(double a) {
 RZ_API double rz_num_sin(double a) {
 	return sin(a);
 }
+
+RZ_API int rz_num_ffs(int a){
+        if (a == 0)
```
Comment (XVilka) [inline librz/util/unum.c]: 1. Please use the clang-format
2. Please add the Doxygen `\brief` for all new API functions.

**Comment by XVilka** [comment]
Could you please rebase or resolve the conflicts? Should be a bit easier after the refactoring you made for `dm` PR. @PulakIIIT

**Comment by MalhotraPulak** [comment]
I have implemented most of the changes from this PR in the `dm` refactor PR already. Glibc heap is compiling on all platforms so we can run Glibc tests on all platforms. I think we can close this PR for now and I will come back to this issue once I am done with Glibc heap viewer in Cutter.

**Pull request closed without merging**